jetty-core/jetty-fcgi/jetty-fcgi-server/src/main/java/org/eclipse/jetty/fcgi/server/internal/ServerFCGIConnection.java

```diff
@@ -407,4 +408,16 @@ public void onFailure(int request, Throwable failure)
             stream = null;
         }
     }
+
+    @Override
+    public void close()
+    {
+        if (stream != null)
+        {
+            Runnable task = stream.getHttpChannel().onFailure(new EofException());
```

Comment:
See comment on HttpConnection... but I care a little less about FCGI

Reply:
I applied the same change I made in `HttpConnection`.